packages/compass-aggregations/src/components/documents-disclosure-menu.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useCallback } from 'react';
+import {
+  Menu,
+  MenuItem,
+  Button,
+  Icon,
+} from '@mongodb-js/compass-components';
+
+export type DocumentsDisclosureOption = 'expanded' | 'collapsed';
+
+export const DocumentsDisclosureMenu: React.FunctionComponent<{
+  onChange: (option: DocumentsDisclosureOption) => void;
+}> = ({
+  onChange
+}) => {
+  const [option, setOption] = useState<DocumentsDisclosureOption>("collapsed");
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const onMenuItemClick = useCallback(
+    (evt) => {
+      const newOption = evt.currentTarget.dataset.action as DocumentsDisclosureOption;
+      evt.stopPropagation();
+      setIsMenuOpen(false);
+      setOption(newOption);
+      onChange(newOption);
```

Comment:
This looks to me like 3 states (2 states and 1 prop kinda) that are all controlling the same thing, I might be missing the reason for doing thins, but it looks to me that you can change the interface to something like `{option: string, onOptionChange: (newOption) => void}` and this should be enough?

Reply:
clean it up and passed down the selected option as a prop.